src/armature/Model.ts

```diff
@@ -169,4 +174,94 @@ export class Model {
 
         return { min, max };
     }
+
+    /**
+     * Given an ambient scene colour, this exports a model in a format traditional 3D software
+     * can read.
+     *
+     * @param {Color} ambientLightColor The scene's ambient component, added to materials.
+     * @returns {{obj: string; mtl: string}} The source code for the .obj file for geometry and
+     * the corresponding .mtl file for materials.
+     */
+    public exportOBJ(ambientLightColor: Color): { obj: string; mtl: string } {
+        const vertices: vec4[] = [];
+        const normals: vec3[] = [];
+        const groups: { indices: number[][]; material: number }[] = [];
+        const materialIndices: Map<BakedMaterial, number> = new Map<BakedMaterial, number>();
+        const materials: BakedMaterial[] = [];
+
+        this.computeRenderInfo(false).geometry.forEach((r: RenderObject) => {
+            // Give each material a unique number representing it. If a material has already been
+            // given a number, don't add it again, ensuring there are no duplicates.
+            if (!materialIndices.has(r.geometry.material)) {
+                materialIndices.set(r.geometry.material, materials.length);
+                materials.push(r.geometry.material);
+            }
+
+            // Add the vertex indices to a group with the specified material
+            groups.push({
+                // Use an array literal and ... to convert the int16 array to doubles
+                indices: chunk(
+                    [...r.geometry.indices].map((i: number) => i + vertices.length + 1),
+                    3
+                ),
+                material: <number>materialIndices.get(r.geometry.material)
+            });
+
+            // Add vertex and normals, transformed into world space
+            vertices.push(
+                ...chunk(r.geometry.vertices, 3).map(([x, y, z]: number[]) =>
+                    vec4.transformMat4(vec4.create(), vec4.fromValues(x, y, z, 1), r.transform)
+                )
+            );
+            normals.push(
+                ...chunk(r.geometry.normals, 3).map(([x, y, z]: number[]) =>
+                    vec3.transformMat3(vec3.create(), vec3.fromValues(x, y, z), r.normalTransform)
+                )
+            );
+        });
+
+        const obj = [
+            'o calderExport',
+
+            // Source the corresponding material file
+            'mtllib calderExport.mtl',
```

Comment:
What if we don't name the material file `calderExport.mtl`? We can probably get away with it for now by always giving it that name, but might as well ask for a file name as well in the `exportOBJ` signature.

Reply:
also, while I'm changing the signature anyway, what are your thoughts on baking in the ambient component? should we just exclude that since I guess we intend to use this for renderers that don't need an approximated ambient light?

Reply:
I think `illum 2` requires ambient, specular, and diffuse components. See the section on Illumination Models at http://paulbourke.net/dataformats/mtl/.

Reply:
I guess I was thinking that if blender adds it even in cycles rendering, then the colors would be weird if we don't just set it to zero, but I don't know if it does or not so I'll just test it to see if it's even a problem

Reply:
ok yeah blender just ignores the Ka parameter, so might as well keep it at non zero for renderers that actually use it